File: portVerifier.c
#include <stdio.h>
#include <netdb.h>

int main(int argc, char *argv[]){
    int mysocket;
    int conn;

    struct sockaddr_in target;
    mysocket = socket(AF_INET, SOCK_STREAM,0); // 0 -> ip protocol
    target.sin_family = AF_INET;
    target.sin_port = htons(argv[2]);
    target.sin_addr.s_addr = inet_addr(argv[1]); 

    conn = connect(mysocket, (struct sockaddr *)&target, sizeof target);
    if(conn == 0){
        printf("port is open\n");
        close(mysocket);
        close(conn);
    }else{
        printf("port is closed\n");
        close(mysocket);
        close(conn);
    }
}

File: py/portVerifier.py
#!/usr/bin/python

import socket,sys

"""
./portVerifier.py <target> <port>
"""

#ip = raw_input("Type your ip :: ")
#port = int(raw_input("Type the port number :: "))

mysocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
if mysocket.connect_ex((sys.argv[1], sys.argv[2])):
	print("Oh.. it's open folks..")
else:
	print(":( It's closed..")


